### ISSUE TITLE
MySQL support for DurableState

### ISSUE THREAD

**Issue by Roiocam** [apache/pekko-persistence-jdbc#174]
## Motivation

supplement #158, continue solves #56

In #158, we notice that other databases support sequence but MySQL does not, but MySQL does have [auto-increment coumn](https://dev.mysql.com/doc/refman/8.3/en/example-auto-increment.html).

I used `SELECT AUTO_INCREMENT FROM information_schema.TABLES` for retrieving recent automatically generated AUTO_INCREMENT value, but the bigger problem is the sequence number only assigned from MySQL, which can not increase them via `SequenceNextValUpdater`.

**Comment by marcopaggioro** [comment]
I was just starting to implement it, but I ran into issues I couldn't handle, and I imagine that's why Akka didn't implement it.

The first issues resolved:
- `VARCHAR tag` in the Postgres schema of the `durable_state` table :arrow_right: `VARCHAR` without a fixed length is not available in MySQL, but it could become `VARCHAR(255)`, as specified in the Oracle and SQLServer schema.
- `state_payload BYTEA` in the Postgres schema of the `durable_state` table :arrow_right: `BYTEA` is not available in MySQL, but it could become `LONGBLOB`.
- `global_offset BIGSERIAL` in the Postgres schema of the `durable_state` table :arrow_right: `BIGSERIAL` (used as` BIGINT` with auto-increment) is not available in MySQL, but it could become `BIGINT AUTO_INCREMENT` and add a `UNIQUE KEY` to `global_offset`.

This could resolve the schema issues.
Moving on to the code, I came across the `SequenceNextValUpdater` classes that require the `getSequenceNextValueExpr` method to extract the next sequence number.

There's no way to get this information in MySQL.
The `LAST_INSERT_ID()` function is only valid within the same session and after an `INSERT`.
Reading the value from `information_schema.TABLES` isn't safe.

That said, I'm not sure what the best approach is.
Currently, updating the durable state requires manually inserting the `global_offset` obtained from `getSequenceNextValueExpr`.
One solution could be to use `REPLACE` instead of `UPDATE` (and `INSERT` too?), which under the hood does `DELETE` (if necessary) + `INSERT`, and the `INSERT` would trigger the auto-increment.
Perhaps I'm missing the point of obtaining the `global_offset` programmatically and not using DBMS increments. Perhaps for legacy versions?

Another alternative is to develop a MariaDB connector that supports sequences and does not have the `getSequenceNextValueExpr` issue.

**Comment by pjfanning** [comment]
@marcopaggioro Yes the issue highlighted in the issue description is the SequenceNextValUpdater that you mentioned.

**Comment by pjfanning** [comment]
@marcopaggioro I'm not an expert on this code but if it was possible to achieve what we need by bypassing the SequenceNextValUpdater - so that the MySQL code path used a completely different approach - that would be ok. The key thing is that we would have test coverage.
